### PR TITLE
Module_2

### PR DESCRIPTION
**Pull request by SemenIlin**
Task1 - task4

### CODE REVIEW

File: module_2_task_1/Taxation.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
+﻿using System;
+
+namespace module_2_task_1
+{
+    class Taxation
+    {
+        private const int _income = 500;
```
Comment (NEDRUT) [inline module_2_task_1/Taxation.cs:7]: Rename const to COMPANY_INCOME

File: module_2_task_1/Program.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            var _tax = 0.0F;
+            var _totalFix = 0.0F;
+
+            while (true)
+            {
+                Console.WriteLine("Input count companies.");
+                taxation.Verify(Console.ReadLine(), ref _countCompanies);
+
+                Console.WriteLine("Input size tax in per cent.");
+                taxation.Verify(Console.ReadLine(), ref _tax);
```
Comment (NEDRUT) [inline module_2_task_1/Program.cs:21]: Console.WriteLine("Input companies count.");
var companiesCount = Console.ReadLine();

Console.WriteLine("Input tax size in per cent.");
var tax = Console.ReadLine();

And then:
var taxation = new Taxation(companiesCount, tax);

File: module_2_task_1/Program.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+                _totalFix = taxation.CalculateTotalTax(_countCompanies, _tax);
```
Comment (NEDRUT) [inline module_2_task_1/Program.cs:23]: var totalFix = taxation.CalculateTotalTax();

File: module_2_task_1/Program.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var taxation = new Taxation();
+
+            var _countCompanies = 0;
+            var _tax = 0.0F;
+            var _totalFix = 0.0F;
```
Comment (NEDRUT) [inline module_2_task_1/Program.cs:13]: Remove

File: module_2_task_1/Taxation.cs [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            {
+                Console.WriteLine(OutputErrorMessage(data));
+                Console.WriteLine("Input value again.");
+
+                data = Console.ReadLine();
+            }
+        }
+        
+        public float CalculateTotalTax(int countCompanies, float tax)
+        {
```
Comment (NEDRUT) [inline module_2_task_1/Taxation.cs:85]: Make Verify methods private and call them here

File: module_2_task_1/Taxation.cs [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        private const int _income = 500;
+
+        private string OutputErrorMessage(string data)
+        {
+            char [] tempArray = data.ToCharArray();
+
+            int countPoint = 0;
+            int countSign = 0;
+
+            foreach (char ch in tempArray)
```
Comment (NEDRUT) [inline module_2_task_1/Taxation.cs:16]: foreach (char ch in data)

File: module_2_task_1/Taxation.cs [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        
+        public float CalculateTotalTax(int countCompanies, float tax)
+        {
+            return (float)countCompanies * _income * tax / 100;
+        }
+        
+        public void Show(float totalFix)
+        {
+            Console.WriteLine($"The total amount of tax = {totalFix}.");
+        }
```
Comment (NEDRUT) [inline module_2_task_1/Taxation.cs:92]: Remove

File: module_2_task_1/Taxation.cs [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+            }
+            else if((countSign == 1) && (countPoint == 1))
+            {
+                return "Input data < 0. " + "Input data include '.' in place ','.";
+            }
+
+            return "Incorrect data.";
+        }    
+
+        public void Verify(string data, ref int value)
```
Comment (NEDRUT) [inline module_2_task_1/Taxation.cs:62]: public void Verify(string data, **out** int value)

File: module_2_task_2/Program.cs [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+                    break;
+                }
+
+                Console.WriteLine();
+            }
+
+            Console.ReadLine();
+        }
+    }
+}
```
Comment (NEDRUT) [inline module_2_task_2/Program.cs:35]: For task2 need the same improvements as for task1

File: module_2_task_2/Age.cs [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+                Console.WriteLine("Input value again.");
+
+                data = Console.ReadLine();
+            }
+        } 
+
+        // Output answer on query.
+        public string GenerateAnswer(int age)
+        {
+            _answerString = "";
```
Comment (NEDRUT) [inline module_2_task_2/Age.cs:76]: Remove

File: module_2_task_3/Program.cs [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+                    break;
+                }
+
+                Console.WriteLine();
+            }
+
+            Console.ReadLine();
+        }
+    }
+}
```
Comment (NEDRUT) [inline module_2_task_3/Program.cs:40]: For task3 need the same improvements as for task1

File: module_2_task_3/ChangeValue.cs [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+        public void Show(float value1, float value2)
+        {
+            Console.WriteLine($"After change first value is {value1}, second value is {value2}.");
+        }
+
+        public void Verify(string data, ref float value)
+        {
+            while (!float.TryParse(data, out value))
+            {
+                Console.WriteLine("Invalid data. May be data include later or '.' in place ','. ");
```
Comment (NEDRUT) [inline module_2_task_3/ChangeValue.cs:25]: TryParse has override which makes possible to parse with . and , simultaneously. Please, use it

File: Module_2/ICalculate.cs [hunk 13]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (NEDRUT) [inline Module_2/ICalculate.cs:5]: Remove unnecessary usings

File: Module_2/Circle.cs [hunk 14]
```diff
@@ -0,0 +1,48 @@
+﻿using System;
+
+namespace Module_2
+{
+    class Circle : ICalculate
+    {
+        private double _radius;
+
+        public Circle() { }
```
Comment (NEDRUT) [inline Module_2/Circle.cs:9]: Remove

File: Module_2/Circle.cs [hunk 15]
```diff
@@ -0,0 +1,48 @@
+﻿using System;
+
+namespace Module_2
+{
+    class Circle : ICalculate
+    {
+        private double _radius;
```
Comment (NEDRUT) [inline Module_2/Circle.cs:7]: readonly

File: Module_2/GetDataFromUser.cs [hunk 16]
```diff
@@ -0,0 +1,423 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Module_2
+{
+    class GetDataFromUser
```
Comment (NEDRUT) [inline Module_2/GetDataFromUser.cs:9]: Wrong class name. Should be noun

File: module_2_task_3/Program.cs [hunk 17]
```diff
@@ -0,0 +1,40 @@
+﻿using System;
+
+namespace module_2_task_3
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var change = new ChangeValue();
```
Comment (NEDRUT) [inline module_2_task_3/Program.cs:9]: Rename class

**Review by NEDRUT — APPROVED** [review]

File: module_2_task_1/Program.cs (hunk 2 above)
Comment (SemenIlin) [reply module_2_task_1/Program.cs:21]: мне нужно создать конструктор, который принимает 2 параметра в первом задании. я правильно понял?

File: module_2_task_3/Program.cs (hunk 17 above)
Comment (SemenIlin) [reply module_2_task_3/Program.cs:9]: Сняли 3 балла за неправильное имя класса, или что-то ещё не так?

**Pull request merged**